jython/test/Python3Test.py3

```diff
@@ -0,0 +1,37 @@
+print ("Python3Test")
+
+import jmri as jmri
+import java
+
+# access constants
+
+if (jmri.Turnout.THROWN != 4) : raise AssertionError('Constant THROWN not right')
+
+# access InstanceManager to get manager
+smc = java.type('jmri.SensorManager')
+sm = jmri.InstanceManager.getNullableDefault(smc)
+
+if (sm == None) : raise AssertionError('No instance manager access')
+
+# use that manager to affect JMRI
+IS1 = sm.provideSensor("IS1")
+
+if (sm.getSensor("IS1") == None) : raise AssertionError('Sensor not created')
+
+# extending a class
+from jmri.jmrit.automat import AbstractAutomaton
+
+class Automat(AbstractAutomaton) :
+    def init(self) :
+        print ("init in Python 3")
+    def handle(self) :
+        print ("handle in Python 3")
+        return False
+Automat().start()
+
+# load the shortcuts
+exec( open("jython/jmri_bindings.py3").read() )
```

Comment:
I asked about the `jython/jmri_bindings.py3` file above. Is this line required in all Python3 scripts to be able to access these constants?

See this line in `jmri.jmrit.logixng.expressions.ExpressionScript`:
`bindings.put("result", result);     // Give the script access to the local variable 'result'`
https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/expressions/ExpressionScript.java#L304

It binds the Jython variable `result` to the Java variable `result`. This is necessary to allow the Jython script to do some calculation and then return a True/False result. Is this not possible with Python3? Or am I missing something?

Reply:
Right now, the built-in bindings don't work for the Python 3 engine.  They get lost somewhere in the wrapper, and I haven't found out where yet (the wrapper is made from some Oracle code, which in turn provides a wrapper, so it's pretty tangled).  So the "bindings.put(..)" call doesn't do anything for Py3.  It's on my list to fix.

Does LogixNG have a way to specify which kind of script engine to invoke?  We should add that too.

Reply:
> Does LogixNG have a way to specify which kind of script engine to invoke? We should add that too.

https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/actions/ActionScript.java#L279
https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/actions/ActionScript.java#L289

https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/expressions/ExpressionScript.java#L312
https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/logixng/expressions/ExpressionScript.java#L322

No, not now but it's easy to add and I'm happy to add it unless you want to do it. What needs to be done is to change the lines above to use the user selected script manager when using one line script and use the filename extension when using a file. It would be good if this is done on the master branch, not the j11master branch.

And then change the swing classes to add a combo box to let the user select the script engine. And update the Xml classes and the schema files to load and store this.

Reply:
There are two separate cases.  For files, the extension should be used to identify the content; if we start allowing people to e.g. "Yes, it's a .js, but I want it to run in jython" craziness will follow.  For code fragments, there needs to be a selector (which in turn needs to be stored and loaded, etc).  

We have jmri.script.swing.ScriptFileChooser for selecting script-specific files, so that's OK.  We need a little Swing widget to do the single-line selection.  Maybe as a JCombobox?  I'll take a look at writing that, but it might be a couple days.

There''s really no hurry on this.  It'll be a while until Py3 is available on Windows, and I don't expect much call for this until then.